dmipy/core/modeling_framework.py

```diff
@@ -1212,8 +1216,24 @@ def fit(self, acquisition_scheme, data,
             start = time()
             mt_fractions = np.empty(
                 np.r_[N_voxels, self.N_models], dtype=float)
+
+            S0_responses = []
```

Comment:
We know what will be the length of `S0_responses`, so it's cleaner to initiate it just with `S0_responses = np.zeros(self.N_models)` or something.